File: .claude/skills/create-worker/references/worker-template.java
// Worker class template (Java) — replace all placeholders before use
// Placeholders: <Name>Worker, TaskTypes.CONSTANT, UseCaseInterface, useCaseMethod, DomainType, Variables.CONSTANT

import io.camunda.client.annotation.JobWorker;
import io.camunda.client.annotation.Variable;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.Map;
import java.util.UUID;

// ---------------------------------------------------------------------------
// Void variant (no output variables — handle returns void):
// ---------------------------------------------------------------------------

@Component
public class <Name>Worker {

    private static final Logger log = LoggerFactory.getLogger(<Name>Worker.class);

    private final UseCaseInterface useCase;

    public <Name>Worker(UseCaseInterface useCase) {
        this.useCase = useCase;
    }

    @JobWorker(type = TaskTypes.CONSTANT)
    public void handle(@Variable UUID subscriptionId) {
        log.debug("Received job for subscriptionId: {}", subscriptionId);
        useCase.useCaseMethod(new DomainType(subscriptionId));
    }
}

// ---------------------------------------------------------------------------
// Return variant (worker produces output variables — handle returns Map<String, Object>):
// ---------------------------------------------------------------------------

@Component
public class <Name>Worker {

    private static final Logger log = LoggerFactory.getLogger(<Name>Worker.class);

    private final UseCaseInterface useCase;

    public <Name>Worker(UseCaseInterface useCase) {
        this.useCase = useCase;
    }

    @JobWorker(type = TaskTypes.CONSTANT)
    public Map<String, Object> handle(@Variable UUID subscriptionId) {
        log.debug("Received job for subscriptionId: {}", subscriptionId);
        useCase.useCaseMethod(new DomainType(subscriptionId));
        return Map.of(Variables.VARIABLE_NAME, value);
    }
}

// Variations:
// @VariableAsType: public void handle(@VariableAsType MyVarsClass variables) { ... }
// Multiple output variables: return Map.of(Variables.A, valueA, Variables.B, valueB);
// Lombok: replace logger field + constructor with @Slf4j and @RequiredArgsConstructor on the class

File: .claude/skills/create-worker/references/worker-template.kt
// Worker class template — replace all placeholders before use
// Placeholders: WorkerName, TaskTypes.CONSTANT, UseCaseInterface, useCaseMethod, DomainType, Variables.CONSTANT

import io.camunda.client.annotation.JobWorker
import io.camunda.client.annotation.Variable
import mu.KotlinLogging
import org.springframework.stereotype.Component
import java.util.UUID

// ---------------------------------------------------------------------------
// Void variant (no output variables — handle returns Unit):
// ---------------------------------------------------------------------------

@Component
class <Name>Worker(
    private val useCase: UseCaseInterface
) {

    private val log = KotlinLogging.logger {}

    @JobWorker(type = TaskTypes.CONSTANT)
    fun handle(@Variable subscriptionId: UUID) {
        log.debug { "Received job for subscriptionId: $subscriptionId" }
        useCase.useCaseMethod(DomainType(subscriptionId))
    }
}

// ---------------------------------------------------------------------------
// Return variant (worker produces output variables — handle returns Map<String, Any>):
// ---------------------------------------------------------------------------

@Component
class <Name>Worker(
    private val useCase: UseCaseInterface
) {

    private val log = KotlinLogging.logger {}

    @JobWorker(type = TaskTypes.CONSTANT)
    fun handle(@Variable subscriptionId: UUID): Map<String, Any> {
        log.debug { "Received job for subscriptionId: $subscriptionId" }
        useCase.useCaseMethod(DomainType(subscriptionId))
        return mapOf(Variables.VARIABLE_NAME to value)
    }
}

// ---------------------------------------------------------------------------
// Dynamic output variant (map contents depend on the use-case result):
// ---------------------------------------------------------------------------

@Component
class <Name>Worker(
    private val useCase: UseCaseInterface
) {

    private val log = KotlinLogging.logger {}

    @JobWorker(type = TaskTypes.CONSTANT)
    fun handle(@Variable subscriptionId: UUID): Map<String, Any> {
        log.debug { "Received job for subscriptionId: $subscriptionId" }
        val result = useCase.useCaseMethod(DomainType(subscriptionId))

        return when (result) {
            null -> mapOf(Variables.FOUND to false)
            else -> mapOf(
                Variables.FOUND to true,
                Variables.RESULT to result.asValue()
            )
        }
    }
}

// Variations:
// @VariableAsType: fun handle(@VariableAsType variables: MyVarsClass) { ... }
// Multiple static output variables: return mapOf(Variables.A to valueA, Variables.B to valueB)